### PR TITLE
new: Entities Table

### PR DESCRIPTION
**Pull request by ktuite**
Getting closer, still more things to refactor.

Screen Shot 2023-02-10 at 12 51 42 PM

### CODE REVIEW

File: src/components/dataset/entity/list.vue [hunk 1]
```diff
@@ -0,0 +1,142 @@
+<!--
```
Comment (ktuite) [inline src/components/dataset/entity/list.vue:1]: all of these components nested in dataset/entities could move up to an entities folder

**Review by ktuite — COMMENTED** [review]
Comments from a discussion with @matthew-white about ideas for reorganizing and reusing code

**Review by ktuite — COMMENTED** [review]
Interactive review with @matthew-white

File: src/routes.js [hunk 2]
```diff
@@ -513,6 +513,18 @@ const routes = [
...
+      }),
+      asyncRoute({
+        path: 'entities',
+        component: 'DatasetEntities',
+        props: true,
+        loading: 'tab',
+        meta: {
+          title: () => [i18n.t('resource.data'), dataset.name],
+          validateData: {
+            project: () => project.permits('entity.list')
```
[the remark this answers is not in the document]
Comment (ktuite) [reply src/routes.js:525]: we have `dataset.list` but not a dataset.read for some reason.

File: test/components/dataset/entities.spec.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  it('shows the download button', async () => {
+    testData.extendedDatasets.createPast(1, { name: 'trees' });
+    testData.extendedEntities.createPast(10);
+    const component = await load(
+      '/projects/1/datasets/trees/entities',
+      { root: false }
+    );
+    const text = component.get('#entity-download-button').text();
+    text.should.equal('Download 10 Entities');
+  });
```
Comment (ktuite) [inline test/components/dataset/entities.spec.js:18]: Copy over tests about odata modal
- does it open
- does it have the correct odata url

File: test/data/entities.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        __id: entityId,
+        __system: {
+          createdAt,
+          creatorId: creator.id.toString(),
+          creatorName: creator.displayName
+        }
+      })
+    };
+  },
+  sort: comparator((submission1, submission2) =>
```
Comment (ktuite) [inline test/data/entities.js:65]: rename submission to entity

File: test/data/entities.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    isBefore(submission2.createdAt, submission1.createdAt))
+});
+
+export const standardSubmissions = view(
+  extendedEntities,
+  omit(['creator'])
+);
```
Comment (ktuite) [inline test/data/entities.js:72]: rename this `standardEntities`

File: test/data/entities.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+// Converts submission response objects to OData. Returns all data even for
```
Comment (ktuite) [inline test/data/entities.js:74]: replace submission with entity and rm comment about encrypted submissions

**Review by ktuite — COMMENTED** [review]
interactive review of tests with @matthew-white

File: src/routes.js [hunk 7]
```diff
@@ -513,6 +513,18 @@ const routes = [
             project: () => project.permits('dataset.list')
           }
         }
+      }),
+      asyncRoute({
```
Comment (ktuite) [inline src/routes.js:517]: test things like 
- title (title meta field)
- if you're a data collector, you can't visit this page (validateData) 
- going from overview to data tab preserves project data (preserveData)
- login loop (requireLogin)

File: test/components/dataset/entities.spec.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+  it('shows the download button', async () => {
+    testData.extendedDatasets.createPast(1, { name: 'trees' });
+    testData.extendedEntities.createPast(10);
+    const component = await load(
+      '/projects/1/datasets/trees/entities',
+      { root: false }
+    );
+    const text = component.get('#entity-download-button').text();
+    text.should.equal('Download 10 Entities');
```
Comment (ktuite) [inline test/components/dataset/entities.spec.js:17]: test href of download button

**Review by matthew-white — APPROVED** [review]

**Pull request merged**